[Automation] native-image run fails for org.jline:jline-reader:3.22.0

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.22.0" ./gradlew clean test -Pcoordinates="org.jline:jline-reader:3.19.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25270673198/job/74092419687
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
rc/org.jline/jline-reader/3.19.0/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.jline/jline-reader/3.19.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.jline/jline/3.21.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.jline/jline-terminal/3.21.0 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=8d988416-fc67-4e37-a3c9-89b25ec5e2b9 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
========================================================================================================================
GraalVM Native Image: Generating 'org.jline.jline-reader_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (5.9s @ 0.17GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                   (32.5s @ 0.89GB)
    8,575 types,   9,675 fields, and  37,414 methods found reachable
    2,717 types,     194 fields, and   1,469 methods registered for reflection
       61 types,      74 fields, and      52 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (4.0s @ 0.99GB)
[4/8] Parsing methods...      [**]                                                                       (4.1s @ 0.63GB)
[5/8] Inlining methods...     [***]                                                                      (2.1s @ 0.68GB)
[6/8] Compiling methods...    [****]                                                                    (18.1s @ 0.84GB)
[7/8] Laying out methods...   [**]                                                                       (3.5s @ 0.77GB)
[8/8] Creating image...       [**]                                                                       (3.5s @ 0.89GB)
   9.95MB (35.77%) for code area:    22,986 compilation units
  14.02MB (50.40%) for image heap:  217,749 objects and 222 resources
   3.85MB (13.83%) for other data
  27.83MB in total image size, 25.76MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   4.69MB java.base                                            2.92MB byte[] for code metadata
   2.09MB svm.jar (Native Image)                               2.61MB byte[] for java.lang.String
 584.05kB assertj-core-3.22.0.jar                              1.39MB java.lang.String
 481.18kB java.xml                                             1.07MB java.lang.Class
 417.88kB jline-reader-3.22.0.jar                            999.11kB byte[] for general heap data
 208.03kB junit-jupiter-engine-5.9.2.jar                     891.80kB com.oracle.svm.core.hub.DynamicHubCompanion
 201.15kB jline-terminal-3.22.0.jar                          496.75kB heap alignment
 120.74kB junit-platform-launcher-1.9.2.jar                  449.50kB java.util.concurrent.ConcurrentHashMap$Node
 103.49kB java.logging                                       355.94kB java.util.HashMap$Node
  99.82kB junit-platform-engine-1.9.2.jar                    280.77kB java.lang.Object[]
 602.95kB for 26 more packages                                 2.57MB for 2123 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 18 component(s) is embedded in binary (948B). 2 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                        5.5s (7.2% of total time) in 759 GCs | Peak RSS: 1.73GB | CPU load: 3.60
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/native/nativeTestCompile/org.jline.jline-reader_tests-tests (executable)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'org.jline.jline-reader_tests-tests' in 1m 16s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_jline.jline_reader.Jline_readerTest > defaultParserHandlesQuotingEscapingAndCompletionMetadata() SUCCESSFUL

org_jline.jline_reader.Jline_readerTest > historyNavigatesPersistsAndExpandsPreviousEvents() FAILED

org_jline.jline_reader.Jline_readerTest > bufferSupportsCursorEditingDeletionAndCopying() SUCCESSFUL

org_jline.jline_reader.Jline_readerTest > completersContributeCandidatesForStringsEnumsAggregateAndArguments() SUCCESSFUL

org_jline.jline_reader.Jline_readerTest > lineReaderReadsScriptedInputWithConfiguredTerminalHistoryAndVariables() FAILED

org_jline.jline_reader.Jline_readerTest > defaultParserReportsIncompleteInputWhenConfiguredForMultilineEditing() SUCCESSFUL

org_jline.jline_reader.Jline_readerTest > keyMapTranslatesDisplaysBindsAndReadsTerminalInput() FAILED

org_jline.jline_reader.Jline_readerTest > defaultHighlighterStylesConfiguredErrorsWithoutChangingDisplayedText() FAILED

org_jline.jline_reader.Jline_readerTest > fileNameCompleterContributesFileAndDirectoryCandidates() FAILED

org_jline.jline_reader.Jline_readerTest > maskingCallbackAndCandidatesExposeDisplayHistoryAndSortingMetadata() SUCCESSFUL


Failures (5):
  JUnit Jupiter:Jline_readerTest:historyNavigatesPersistsAndExpandsPreviousEvents()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'historyNavigatesPersistsAndExpandsPreviousEvents', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.historyNavigatesPersistsAndExpandsPreviousEvents(Jline_readerTest.java:174)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]
  JUnit Jupiter:Jline_readerTest:lineReaderReadsScriptedInputWithConfiguredTerminalHistoryAndVariables()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'lineReaderReadsScriptedInputWithConfiguredTerminalHistoryAndVariables', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.lineReaderReadsScriptedInputWithConfiguredTerminalHistoryAndVariables(Jline_readerTest.java:216)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]
  JUnit Jupiter:Jline_readerTest:keyMapTranslatesDisplaysBindsAndReadsTerminalInput()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'keyMapTranslatesDisplaysBindsAndReadsTerminalInput', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.keyMapTranslatesDisplaysBindsAndReadsTerminalInput(Jline_readerTest.java:267)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]
  JUnit Jupiter:Jline_readerTest:defaultHighlighterStylesConfiguredErrorsWithoutChangingDisplayedText()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'defaultHighlighterStylesConfiguredErrorsWithoutChangingDisplayedText', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.defaultHighlighterStylesConfiguredErrorsWithoutChangingDisplayedText(Jline_readerTest.java:342)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]
  JUnit Jupiter:Jline_readerTest:fileNameCompleterContributesFileAndDirectoryCandidates()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'fileNameCompleterContributesFileAndDirectoryCandidates', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.fileNameCompleterContributesFileAndDirectoryCandidates(Jline_readerTest.java:295)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]

Test run finished after 9 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[        10 tests found           ]
[         0 tests skipped         ]
[        10 tests started         ]
[         0 tests aborted         ]
[         5 tests successful      ]
[         5 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/native/nativeTestCompile/org.jline.jline-reader_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 19s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jline:jline-reader:3.19.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 21s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 3.22.0 failed with exit code 1.
FAILED[native-image run][3.22.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.22.0" ./gradlew clean test -Pcoordinates="org.jline:jline-reader:3.19.0"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.22.0" ./gradlew clean test -Pcoordinates="org.jline:jline-reader:3.19.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25270673198/job/74092419742
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (35.1s @ 0.88GiB)
    8,698 types,   9,724 fields, and  37,829 methods found reachable
      287 types,      83 fields, and     717 methods registered for reflection
       61 types,      74 fields, and      52 methods registered for JNI access
      222 resource accesses registered with 554.92KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.2s @ 0.97GiB)
[4/8] Parsing methods...      [**]                                                                      (4.1s @ 0.58GiB)
[5/8] Inlining methods...     [***]                                                                     (2.0s @ 0.63GiB)
[6/8] Compiling methods...    [****]                                                                   (19.5s @ 0.70GiB)
[7/8] Laying out methods...   [**]                                                                      (2.9s @ 0.85GiB)
[8/8] Creating image...       [**]                                                                      (3.3s @ 0.97GiB)
   8.80MiB (33.93%) for code area:    23,247 compilation units
  13.38MiB (51.57%) for image heap:  222,423 objects and 92 resources
   3.76MiB (14.50%) for other data
  25.94MiB in total image size, 23.94MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
 945.61KiB o.g.n.runtime/com.oracle.svm.core.code              2.74MiB byte[] for code metadata
 725.79KiB java.base/java.util                                 2.58MiB byte[] for string data
 408.37KiB java.base/jdk.internal.classfile.impl               1.39MiB java.base/java.lang.String
 336.15KiB org.jline.reader.impl                               1.03MiB java.base/java.lang.Class
 332.96KiB java.base/java.lang                               951.34KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 245.27KiB java.base/java.util.concurrent                    556.34KiB byte[] for embedded resources
 228.94KiB java.base/java.text                               556.28KiB byte[] for general heap data
 224.55KiB java.base/com.sun.crypto.provider                 352.34KiB java.base/java.util.HashMap$Node
 223.93KiB org.assertj.core.internal.bytebuddy.jar.asm       337.11KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 154.86KiB java.base/java.io                                 298.89KiB byte[] for reflection metadata
   5.06MiB for 320 more packages                               2.65MiB for 2135 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 18 component(s) is embedded in binary (993B). 2 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       5.3s (6.5% of total time) in 760 GCs | Peak RSS: 1.61GiB | CPU load: 3.61
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/native/nativeTestCompile/org.jline.jline-reader_tests-tests (executable, 23.94MiB)
 /tmp/native-image-build-output.json (build_info, 1.03KiB)
========================================================================================================================
Finished generating 'org.jline.jline-reader_tests-tests' in 1m 19s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_jline.jline_reader.Jline_readerTest > defaultParserHandlesQuotingEscapingAndCompletionMetadata() SUCCESSFUL

org_jline.jline_reader.Jline_readerTest > historyNavigatesPersistsAndExpandsPreviousEvents() FAILED

org_jline.jline_reader.Jline_readerTest > bufferSupportsCursorEditingDeletionAndCopying() SUCCESSFUL

org_jline.jline_reader.Jline_readerTest > completersContributeCandidatesForStringsEnumsAggregateAndArguments() SUCCESSFUL

org_jline.jline_reader.Jline_readerTest > lineReaderReadsScriptedInputWithConfiguredTerminalHistoryAndVariables() FAILED

org_jline.jline_reader.Jline_readerTest > defaultParserReportsIncompleteInputWhenConfiguredForMultilineEditing() SUCCESSFUL

org_jline.jline_reader.Jline_readerTest > keyMapTranslatesDisplaysBindsAndReadsTerminalInput() FAILED

org_jline.jline_reader.Jline_readerTest > defaultHighlighterStylesConfiguredErrorsWithoutChangingDisplayedText() FAILED

org_jline.jline_reader.Jline_readerTest > fileNameCompleterContributesFileAndDirectoryCandidates() FAILED

org_jline.jline_reader.Jline_readerTest > maskingCallbackAndCandidatesExposeDisplayHistoryAndSortingMetadata() SUCCESSFUL


Failures (5):
  JUnit Jupiter:Jline_readerTest:historyNavigatesPersistsAndExpandsPreviousEvents()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'historyNavigatesPersistsAndExpandsPreviousEvents', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.historyNavigatesPersistsAndExpandsPreviousEvents(Jline_readerTest.java:174)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]
  JUnit Jupiter:Jline_readerTest:lineReaderReadsScriptedInputWithConfiguredTerminalHistoryAndVariables()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'lineReaderReadsScriptedInputWithConfiguredTerminalHistoryAndVariables', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.lineReaderReadsScriptedInputWithConfiguredTerminalHistoryAndVariables(Jline_readerTest.java:216)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]
  JUnit Jupiter:Jline_readerTest:keyMapTranslatesDisplaysBindsAndReadsTerminalInput()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'keyMapTranslatesDisplaysBindsAndReadsTerminalInput', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.keyMapTranslatesDisplaysBindsAndReadsTerminalInput(Jline_readerTest.java:267)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]
  JUnit Jupiter:Jline_readerTest:defaultHighlighterStylesConfiguredErrorsWithoutChangingDisplayedText()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'defaultHighlighterStylesConfiguredErrorsWithoutChangingDisplayedText', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.defaultHighlighterStylesConfiguredErrorsWithoutChangingDisplayedText(Jline_readerTest.java:342)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]
  JUnit Jupiter:Jline_readerTest:fileNameCompleterContributesFileAndDirectoryCandidates()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'fileNameCompleterContributesFileAndDirectoryCandidates', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.fileNameCompleterContributesFileAndDirectoryCandidates(Jline_readerTest.java:295)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]

Test run finished after 14 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[        10 tests found           ]
[         0 tests skipped         ]
[        10 tests started         ]
[         0 tests aborted         ]
[         5 tests successful      ]
[         5 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/native/nativeTestCompile/org.jline.jline-reader_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 23s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jline:jline-reader:3.19.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 25s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 3.22.0 failed with exit code 1.
FAILED[native-image run][3.22.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.22.0" ./gradlew clean test -Pcoordinates="org.jline:jline-reader:3.19.0"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.22.0" ./gradlew clean test -Pcoordinates="org.jline:jline-reader:3.19.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25270673198/job/74092419649
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (40.2s @ 0.97GiB)
    9,230 types,  10,520 fields, and  40,842 methods found reachable
      415 types,      83 fields, and     918 methods registered for reflection
       61 types,      74 fields, and      52 methods registered for JNI access
      224 resource accesses registered with 555.05KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.8s @ 1.07GiB)
[4/8] Parsing methods...      [**]                                                                      (4.6s @ 0.76GiB)
[5/8] Inlining methods...     [***]                                                                     (2.1s @ 0.83GiB)
[6/8] Compiling methods...    [*****]                                                                  (22.2s @ 0.84GiB)
[7/8] Laying out methods...   [**]                                                                      (3.4s @ 1.03GiB)
[8/8] Creating image...       [**]                                                                      (3.9s @ 1.18GiB)
   9.92MiB (33.50%) for code area:    25,550 compilation units
  15.69MiB (52.98%) for image heap:  260,885 objects and 94 resources
   4.00MiB (13.52%) for other data
  29.61MiB in total image size, 27.44MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.05MiB o.g.n.runtime/com.oracle.svm.core.code              3.09MiB byte[] for code metadata
 732.82KiB java.base/java.util                                 2.87MiB byte[] for string data
 408.34KiB java.base/jdk.internal.classfile.impl               1.53MiB java.base/java.lang.String
 341.87KiB java.base/java.lang                                 1.10MiB java.base/java.lang.Class
 336.15KiB org.jline.reader.impl                               0.99MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 267.01KiB java.base/com.sun.crypto.provider                 912.79KiB byte[] for general heap data
 245.39KiB java.base/java.util.concurrent                    556.51KiB byte[] for embedded resources
 231.51KiB java.base/java.text                               382.80KiB heap alignment
 223.93KiB org.assertj.core.internal.bytebuddy.jar.asm       366.68KiB byte[] for reflection metadata
 192.83KiB java.base/sun.security.provider                   363.05KiB java.base/java.lang.Object[]
   5.96MiB for 330 more packages                               3.59MiB for 2284 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 18 component(s) is embedded in binary (993B). 2 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       6.3s (6.9% of total time) in 853 GCs | Peak RSS: 1.81GiB | CPU load: 3.62
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/native/nativeTestCompile/org.jline.jline-reader_tests-tests (executable, 27.44MiB)
 /tmp/native-image-build-output.json (build_info, 1.03KiB)
========================================================================================================================
Finished generating 'org.jline.jline-reader_tests-tests' in 1m 30s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_jline.jline_reader.Jline_readerTest > defaultParserHandlesQuotingEscapingAndCompletionMetadata() SUCCESSFUL

org_jline.jline_reader.Jline_readerTest > historyNavigatesPersistsAndExpandsPreviousEvents() FAILED

org_jline.jline_reader.Jline_readerTest > bufferSupportsCursorEditingDeletionAndCopying() SUCCESSFUL

org_jline.jline_reader.Jline_readerTest > completersContributeCandidatesForStringsEnumsAggregateAndArguments() SUCCESSFUL

org_jline.jline_reader.Jline_readerTest > lineReaderReadsScriptedInputWithConfiguredTerminalHistoryAndVariables() FAILED

org_jline.jline_reader.Jline_readerTest > defaultParserReportsIncompleteInputWhenConfiguredForMultilineEditing() SUCCESSFUL

org_jline.jline_reader.Jline_readerTest > keyMapTranslatesDisplaysBindsAndReadsTerminalInput() FAILED

org_jline.jline_reader.Jline_readerTest > defaultHighlighterStylesConfiguredErrorsWithoutChangingDisplayedText() FAILED

org_jline.jline_reader.Jline_readerTest > fileNameCompleterContributesFileAndDirectoryCandidates() FAILED

org_jline.jline_reader.Jline_readerTest > maskingCallbackAndCandidatesExposeDisplayHistoryAndSortingMetadata() SUCCESSFUL


Failures (5):
  JUnit Jupiter:Jline_readerTest:historyNavigatesPersistsAndExpandsPreviousEvents()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'historyNavigatesPersistsAndExpandsPreviousEvents', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.historyNavigatesPersistsAndExpandsPreviousEvents(Jline_readerTest.java:174)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]
  JUnit Jupiter:Jline_readerTest:lineReaderReadsScriptedInputWithConfiguredTerminalHistoryAndVariables()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'lineReaderReadsScriptedInputWithConfiguredTerminalHistoryAndVariables', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.lineReaderReadsScriptedInputWithConfiguredTerminalHistoryAndVariables(Jline_readerTest.java:216)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]
  JUnit Jupiter:Jline_readerTest:keyMapTranslatesDisplaysBindsAndReadsTerminalInput()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'keyMapTranslatesDisplaysBindsAndReadsTerminalInput', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.keyMapTranslatesDisplaysBindsAndReadsTerminalInput(Jline_readerTest.java:267)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]
  JUnit Jupiter:Jline_readerTest:defaultHighlighterStylesConfiguredErrorsWithoutChangingDisplayedText()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'defaultHighlighterStylesConfiguredErrorsWithoutChangingDisplayedText', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.defaultHighlighterStylesConfiguredErrorsWithoutChangingDisplayedText(Jline_readerTest.java:342)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]
  JUnit Jupiter:Jline_readerTest:fileNameCompleterContributesFileAndDirectoryCandidates()
    MethodSource [className = 'org_jline.jline_reader.Jline_readerTest', methodName = 'fileNameCompleterContributesFileAndDirectoryCandidates', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unable to create a terminal
       org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:363)
       org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:303)
       org_jline.jline_reader.Jline_readerTest.fileNameCompleterContributesFileAndDirectoryCandidates(Jline_readerTest.java:295)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jna
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:367)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider jansi
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:376)
         [...]
       Suppressed: java.io.IOException: Unable to find terminal provider exec
         org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:68)
         org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:386)
         [...]

Test run finished after 8 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[        10 tests found           ]
[         0 tests skipped         ]
[        10 tests started         ]
[         0 tests aborted         ]
[         5 tests successful      ]
[         5 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jline/jline-reader/3.19.0/build/native/nativeTestCompile/org.jline.jline-reader_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 34s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jline:jline-reader:3.19.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 36s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 3.22.0 failed with exit code 1.
FAILED[native-image run][3.22.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.22.0" ./gradlew clean test -Pcoordinates="org.jline:jline-reader:3.19.0"]
```

--- a/metadata/org.jline/jline-reader/3.22.0/reachability-metadata.json
+++ b/metadata/org.jline/jline-reader/3.22.0/reachability-metadata.json
@@ -1,0 +1,48 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "org.jline.terminal.spi.TerminalProvider"
+      },
+      "type": "org.jline.terminal.impl.exec.ExecTerminalProvider",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "org.jline.terminal.spi.TerminalProvider"
+      },
+      "glob": "META-INF/services/org/jline/terminal/provider/exec"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jline.terminal.spi.TerminalProvider"
+      },
+      "glob": "META-INF/services/org/jline/terminal/provider/jansi"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jline.terminal.spi.TerminalProvider"
+      },
+      "glob": "META-INF/services/org/jline/terminal/provider/jna"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jline.utils.InfoCmp"
+      },
+      "glob": "org/jline/utils/capabilities.txt"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jline.utils.InfoCmp"
+      },
+      "glob": "org/jline/utils/xterm-256color.caps"
+    }
+  ]
+}

--- a/metadata/org.jline/jline-reader/index.json
+++ b/metadata/org.jline/jline-reader/index.json
@@ -1,6 +1,21 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "3.22.0",
+    "test-version" : "3.19.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/jline/jline-reader/$version$/jline-reader-$version$-sources.jar",
+    "repository-url" : "https://github.com/jline/jline3",
+    "test-code-url" : "https://github.com/jline/jline3/tree/jline-parent-$version$/reader/src/test",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/org/jline/jline-reader/$version$/jline-reader-$version$-javadoc.jar",
+    "description" : "JLine Reader provides the line-editing, history, completion, key binding, and prompt-reading APIs used to build interactive command-line interfaces on the JVM. It sits on top of JLine terminal abstractions so Java applications can implement shell-like input behavior across supported consoles and terminal backends.",
+    "tested-versions" : [
+      "3.22.0"
+    ],
+    "allowed-packages" : [
+      "org.jline"
+    ]
+  },
+  {
     "metadata-version" : "3.19.0",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/jline/jline-reader/$version$/jline-reader-$version$-sources.jar",
     "repository-url" : "https://github.com/jline/jline3",

--- a/stats/org.jline/jline-reader/3.22.0/stats.json
+++ b/stats/org.jline/jline-reader/3.22.0/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "3.22.0",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 9638,
+        "missed" : 17677,
+        "ratio" : 0.352846,
+        "total" : 27315
+      },
+      "line" : {
+        "covered" : 1814,
+        "missed" : 3659,
+        "ratio" : 0.331445,
+        "total" : 5473
+      },
+      "method" : {
+        "covered" : 307,
+        "missed" : 533,
+        "ratio" : 0.365476,
+        "total" : 840
+      }
+    }
+  } ]
+}

